src/commands/configuration/config-subcommands/reactionroles.ts

```diff
@@ -0,0 +1,862 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) 2023-present Mirage Aegis
+ * 
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ * 
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ * 
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+import { 
+    EmbedBuilder, ChatInputCommandInteraction, SlashCommandSubcommandBuilder,
+    StringSelectMenuBuilder, StringSelectMenuOptionBuilder, ActionRowBuilder,
+    InteractionResponse, CollectorFilter, StringSelectMenuInteraction, ComponentType,
+    ButtonBuilder, ButtonStyle, MessageComponentInteraction, Message, MessageReaction,
+    User, Snowflake, Collection, RoleSelectMenuBuilder, Emoji, RoleSelectMenuInteraction,
+    Role, ButtonInteraction
+} from "discord.js";
+import { Subcommand } from "../../../util/command-template.js";
+import { defaultErrorHandler } from "../../../util/error-handler.js";
+import {
+    CONFIRM_RR, REGULAR_RR, ReactionRoleStyle, Server, UNIQUE_RR
+} from "../../../schemas/server.js";
+import {
+    AZURE,
+    BLUE, GREEN, GREY, MAGENTA, ORANGE, PINK, PURPLE, RED, TEAL, YELLOW
+} from "../../../util/colours.js";
+
+/*
+ * Reaction role setup command for creating reaction roles.
+ */
+
+type MessageTypeMenu = "plain" | "embed";
+
+const PLAIN: MessageTypeMenu = "plain";
+const EMBED: MessageTypeMenu = "embed";
+
+const SELECT_MENU_TIMEOUT: number = 60_000;
+const PARAGRAPH_TIMEOUT: number = 300_000;
+const REACTION_TIMEOUT: number = 60_000;
+
+const BUTTONS_PER_ROW: number = 5;
+const MIN_RR: number = 1;
+const MAX_RR: number = 15;
+
+const name: string = "reactionroles";
+
+export const command: Subcommand = {
+    // Command headers
+    data: new SlashCommandSubcommandBuilder()
+        .setName(name)
+        .setDescription("Starts a reaction roles setup wizard"),
```

Comment:
The `/config reactionroles` command is a lengthy one that walks users through the process of creating reaction roles.

It consists of seven steps:
1. Choosing a reaction role style.
2. Choosing a message format (plain text or embedded message).
3. Creating the message.
4. Creating reaction roles.
5. Reviewing the message and reaction roles.
6. Sending the reaction roles message.
7. Saving the data to the database.